File: TareaSemana4/AgendaClinica.cs
using System;
using System.Collections.Generic;

namespace AgendaClinicaApp
{
    public class AgendaClinica
    {
        private List<Paciente> listaPacientes;

        public AgendaClinica()
        {
            listaPacientes = new List<Paciente>();
        }

        public void AgregarTurno(Paciente paciente)
        {
            listaPacientes.Add(paciente);
            Console.WriteLine("\n✔ Turno registrado exitosamente.");
        }

        public void ConsultarTurnos()
        {
            if (listaPacientes.Count == 0)
            {
                Console.WriteLine("\n⚠ No hay turnos registrados.");
                return;
            }

            Console.WriteLine("\n📋 Lista de turnos registrados:");
            foreach (var p in listaPacientes)
            {
                Console.WriteLine($"🩺 {p.Nombre} | Cédula: {p.Cedula} | Fecha: {p.FechaTurno} | Hora: {p.HoraTurno} | Motivo: {p.Motivo}");
            }
        }

        public void BuscarPorCedula(string cedula)
        {
            bool encontrado = false;

            foreach (var p in listaPacientes)
            {
                if (p.Cedula == cedula)
                {
                    Console.WriteLine($"\n🔍 Turno encontrado: {p.Nombre}, {p.FechaTurno}, {p.HoraTurno}, Motivo: {p.Motivo}");
                    encontrado = true;
                }
            }

            if (!encontrado)
            {
                Console.WriteLine("\n❌ No se encontró un turno con esa cédula.");
            }
        }
    }
}

File: TareaSemana4/AgendaClinica.java
public class AgendaClinica {
    private Paciente[] pacientes;
    private int contador;

    public AgendaClinica(int capacidad) {
        pacientes = new Paciente[capacidad];
        contador = 0;
    }

    public void agregarTurno(Paciente p) {
        if (contador < pacientes.length) {
            pacientes[contador++] = p;
            System.out.println("Turno agregado exitosamente.\n");
        } else {
            System.out.println("No se pueden agregar más turnos. Capacidad llena.");
        }
    }

    public void mostrarTurnos() {
        System.out.println("\n--- Turnos Registrados ---");
        for (int i = 0; i < contador; i++) {
            pacientes[i].mostrarDatos();
        }
    }

    public void buscarTurno(String cedula) {
        boolean encontrado = false;
        for (int i = 0; i < contador; i++) {
            if (pacientes[i].getCedula().equals(cedula)) {
                System.out.println("Turno encontrado:");
                pacientes[i].mostrarDatos();
                encontrado = true;
                break;
            }
        }
        if (!encontrado) {
            System.out.println("No se encontró ningún turno con esa cédula.\n");
        }
    }
}
